integration/fuse/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java

```diff
@@ -132,70 +137,147 @@ public void umount() {
   }
 
   public int openCallback(String path, ByteBuffer buf) {
-    FuseFileInfo fi = new FuseFileInfo(buf);
-    return open(path, fi);
+    try {
+      return open(path, FuseFileInfo.wrap(buf));
+    } catch (Exception e) {
+      LOG.error("Failed to open {}: ", path, e);
```

Comment:
Something to be careful of is that if you want to log the exception with `LOG.error(..., e)`, it doesn't work with the `{}` substitution ability. So, that means all of the log lines with exception as the last parameter should look something like:
```
LOG.error(String.format("... %s", path), e)
```

This way, the log message has all the parameters, and the exception and stack is printed.

Reply:
@gpang this is actually unnecessary to use `String.format`. https://stackoverflow.com/a/51781806/3161750
As long as you pass a Throwable as the last arg, the full log of exception will be printed as an undocumented feature.
I've been leveraging this feature for a long time